### ISSUE TITLE
"Couldn't update the settings. Check your connection"

### ISSUE THREAD

**Issue by asturel** [actions-on-google/actionssdk-smart-home-nodejs#3]
After I choose the "device" and login there's a progressdialog with "Linking your ... account".. then I got an error: "Couldn't update the settings. Check your connection"
(Used the mock server)

```
POST /token 200 17.212 ms - 100
POST / 404 1.183 ms - 140

```
Probably the 404 reply for / ?

**Comment by asturel** [comment]
Sorry, I see i have to implement sync, query, exec

**Comment by Kant73** [comment]
Hi, I have the same problem.
After adding my device provider (click on [test] your_action_name and then login), there's a progressdialog with "Linking your ... account".. then I got an error: "Couldn't update the settings. Check your connection"
Last week it worked without any issue (just cloning the repo, no other modifications)
What have you done to make it work ?
Maybe @smishra2 can help me too :)
Thanks in advance ;)

**Comment by asturel** [comment]
check your action.json url (make sure it has "/ha" querystring) -- i dont see any POST to / (thats called when its updating..)

**Comment by dooyawoo** [comment]
@Kant73 Hi, I have the same problem.What have you done to make it work ?

**Comment by Kant73** [comment]
@dooyawoo Hi, I don't really remember what I've done but if I'm not wrong it was just a configuration problem, I will check it on Tuesday when I return to work 😉

**Comment by pantongren** [comment]
@Kant73  Brother, have  you solved this issue?

**Comment by Kant73** [comment]
@pantongren I'm not sure that it was that but after checking my git repo , I have delete the `"deviceControl": {
    },
` in the `action.json` config file because it's not needed anymore (the doc wasn't up to date when I was looking at it 2 month ago, hope that it will help you.

**Comment by pantongren** [comment]
@Kant73 thank you brother, but  it still doesn't work.Did you add something like data source to this given sample? I added a set of data to the "Data" constant in the datastore.js file, if not, it would report a null object refrence exception. Did you add something to the given sample or do some modifications?

**Comment by Kant73** [comment]
@pantongren Sorry but no, my Data variable has no modification: `const Data = {};`

**Comment by JasonAyla** [comment]
@asturel Google would fire a SYNC request to your service to get devices, you should at least return 200 for this request.

**Comment by sunjunlishi** [comment]
i have one account ok. but when change to another account , i  have the same question?????
why????

**Comment by Fleker** [comment]
So:
Account A&B: You can link your account but don't get a sync?

**Comment by JasonAyla** [comment]
@Fleker Google would automatically send a SYNC request to your webhook after you have finished the account linking.

**Comment by sunjunlishi** [comment]
sunjunlishi@gmail.com is ok.  I use 
![g1](https://user-images.githubusercontent.com/11845664/30528550-b43b4250-9c66-11e7-9824-ec4df06d04cd.png)
![g2](https://user-images.githubusercontent.com/11845664/30528555-bf2081da-9c66-11e7-8073-727d057b890a.png)
![g3](https://user-images.githubusercontent.com/11845664/30528558-c1ed128e-9c66-11e7-975a-b50703adebc6.png)


when i use another xingteliot@gmail.com  when response say ,"could not update setting,checking your connection" and no device sync  event

**Comment by sunjunlishi** [comment]
xingteliot@gmail.com not linking account ok.
and sunjunlishi@gmail.com all is ok.sync 、exeute ,and so on all is ok.

**Comment by sunjunlishi** [comment]
and  i  use your example again, error is  same.

**Comment by Fleker** [comment]
@sunjunlishi how are you setting up the project? Are you using one project and sharing with the second account, or creating one project for each account?

**Comment by sunjunlishi** [comment]
@Fleker  creating one project for each account

**Comment by Fleker** [comment]
@sunjunlishi I've filed an issue internally and will check out what's going on.

**Comment by sunjunlishi** [comment]
@Fleker authorization code type you have got it,thanks, all my accounts is ok.
but  implicit type, is not ok. Anyway, it has worked and the problem has been solved for me

**Comment by Fleker** [comment]
@sunjunlishi I do not entirely understand. Can you provide more details into the issue you encountered and how you managed to fix it?

**Comment by sunjunlishi** [comment]
oauth2 have two types,one is authorization code type,you have resolve it
![a1](https://user-images.githubusercontent.com/11845664/30622555-efd552c2-9de4-11e7-847c-a4757c333b7f.jpg)
Anyway, it has worked and the problem has been solved for me.

but the other type implicit,i think  about some accounts not be used still.

**Comment by sunjunlishi** [comment]
ok,bad. i create a new acount xingtelgroup@gmail.com and it failed

**Comment by Fleker** [comment]
I don't think implicit auth works for Smart Home projects at all.

**Comment by dziedzic99** [comment]
I am still experiencing this issue both using my own page and ngrok. My code and sample are both not working. Any solutions found? I am using Authorization code instead of Implicit mode. Is this all right?

**Comment by Fleker** [comment]
@dziedzic99 are you using an Android phone? Can you pull logs during this point with `adb logcat`? What you have should be working fine. It may be a server-side issue.

**Comment by dziedzic99** [comment]
@Fleker
You mean Google server or my server?
Also I will do adb debug later when I have time.

**Comment by Fleker** [comment]
Not sure yet

**Comment by dziedzic99** [comment]
```
10-30 20:01:04.650  4618  4631 I ActivityManager: START u0 {cmp=com.google.android.googlequicksearchbox/com.google.android.apps.gsa.assistant.handoff.AssistantHandoffActivity (has extras)} from uid 10029
10-30 20:01:04.712  4857  4857 W AddInstrumentAdapter: Handoff activity for add instrument for transactions is disabled with a flag
10-30 20:01:04.712  4857  4857 W PaymentsAuthAdapter: Handoff activity for payments auth is disabled with a flag
10-30 20:01:04.712  4857  4857 W CvcChallengeAdapter: Handoff activity for CVC challenge for transactions is disabled with a flag
10-30 20:01:04.712  4857  4857 W FixInstrumentAdapter: Handoff activity for fix instrument for transactions is disabled with a flag
10-30 20:01:04.713  4618 10843 I ActivityManager: START u0 {flg=0x800000 cmp=com.google.android.googlequicksearchbox/com.google.android.apps.gsa.assistant.handoff.BrowserControlActivity (has extras)} from uid 10029
10-30 20:01:04.751  4618 11207 I ActivityManager: START u0 {act=android.intent.action.VIEW dat=https://assistant.google.com/... flg=0x40808000 cmp=com.android.chrome/com.google.android.apps.chrome.Main (has extras)} from uid 10029
10-30 20:01:04.803  4618 21856 I ActivityManager: START u0 {act=android.intent.action.VIEW dat=customtab://350e48e8-70e3-4ed9-bfd7-0b770cc13016?https://assistant.google.com/services/auth/handoffs/auth/start?provider=billboard-clock_dev&return_url=assistant-handoff://complete&user_email=jandziedzic99@gmail.com&redirect_uri=assistant-handoff%3A%2F%2Fcomplete flg=0x58088000 cmp=com.android.chrome/org.chromium.chrome.browser.customtabs.SeparateTaskCustomTabActivity (has extras)} from uid 10039
10-30 20:01:04.859  7981  8003 W cr_tabmodel: Starting to fetch tab list.
10-30 20:01:04.860  7981  8003 E cr_tabmodel: State file does not exist.
10-30 20:01:04.887  4618  4950 I WindowManager: Failed to capture screenshot of Token{f83171e ActivityRecord{a164f59 u0 com.google.android.googlequicksearchbox/com.google.android.apps.gsa.assistant.handoff.BrowserControlActivity t23668}} appWin=Window{bd20093 u0 com.google.android.googlequicksearchbox/com.google.android.apps.gsa.assistant.handoff.BrowserControlActivity} drawState=4
10-30 20:01:04.945  4857  5070 D OpenGLRenderer: endAllActiveAnimators on 0x7641a49000 (RippleDrawable) with handle 0x764227d200
10-30 20:01:05.059  4618  4737 I ActivityManager: Displayed com.android.chrome/org.chromium.chrome.browser.customtabs.SeparateTaskCustomTabActivity: +205ms (total +20s320ms)
10-30 20:01:05.729  7981  7981 D SharedPreferencesImpl: Time required to fsync /data/user/0/com.android.chrome/shared_prefs/com.android.chrome_preferences.xml: [<1: 0, <2: 0, <4: 1, <8: 47, <16: 25, <32: 1, <64: 4, <128: 7, <256: 1, <512: 0, <1024: 1, <2048: 0, <4096: 0, <8192: 0, <16384: 0, >=16384: 0]
10-30 20:01:05.730  7981  7981 D QueuedWork: waited: [<1: 0, <2: 18, <4: 1, <8: 3, <16: 7, <32: 1, <64: 0, <128: 0, <256: 0, <512: 0, <1024: 1, <2048: 0, <4096: 0, <8192: 0, <16384: 0, >=16384: 0]
10-30 20:01:05.732  7981  7981 I Choreographer: Skipped 35 frames!  The application may be doing too much work on its main thread.
10-30 20:01:07.204  4618 11207 I ActivityManager: START u0 {act=android.intent.action.VIEW cat=[android.intent.category.BROWSABLE] dat=assistant-handoff://complete/?result_code=SUCCESS&result_message=Accounts+now+linked. flg=0x10000000 cmp=com.google.android.googlequicksearchbox/com.google.android.apps.gsa.assistant.handoff.BrowserReturnActivity (has extras)} from uid 10039
10-30 20:01:07.210  4618 21857 I ActivityManager: moveTaskToBack: TaskRecord{acc3590 #23682 I=com.android.chrome/org.chromium.chrome.browser.customtabs.SeparateTaskCustomTabActivity U=0 StackId=1 sz=1}
10-30 20:01:07.258  4618  8607 I MediaFocusControl:  AudioFocus  abandonAudioFocus() from uid/pid 10039/7981 clientId=android.media.AudioManager@c623e11org.chromium.content.browser.AudioFocusDelegate@8cdd060
10-30 20:01:07.269  4618 11207 I ActivityManager: Process com.android.chrome:sandboxed_process0 (pid 8416) has died: fore TOP 
10-30 20:01:07.269  4618 11207 W ActivityManager: Scheduling restart of crashed service com.android.chrome/org.chromium.content.app.SandboxedProcessService0 in 1000ms
10-30 20:01:07.271  4618  4680 W zygote64: Failed to open process cgroup uid 99319 pid 8416: No such file or directory
10-30 20:01:07.271  4618  4680 E zygote64: Error encountered killing process cgroup uid 99319 pid 8416: No such file or directory
10-30 20:01:07.277  4618  4719 D KernelUidCpuTimeReader: Removing uids 99319-99319
10-30 20:01:07.288  4618 10699 I WifiService: getConnectionInfo uid=10012
10-30 20:01:07.289  4618 11208 I WifiService: getConnectionInfo uid=10012
10-30 20:01:07.292  4618  7464 I WifiService: getDhcpInfo uid=10012
10-30 20:01:07.300  4618 10841 I WifiService: getConnectionInfo uid=10012
10-30 20:01:07.304  4618 10843 I WifiService: getDhcpInfo uid=10012
10-30 20:01:07.307  4618  7458 I ActivityManager: START u0 {dat=assistant-handoff://complete/?result_code=SUCCESS&result_message=Accounts+now+linked. cmp=com.google.android.googlequicksearchbox/com.google.android.apps.gsa.assistant.handoff.BrowserControlActivity} from uid 10029
10-30 20:01:07.331  3726  3817 E QC-QMI  : linux_qmi_qmux_io_wake_lock: Err in writing wakelock=qmuxd_port_wl_0, error [1:Operation not permitted]
10-30 20:01:07.331  3726  3817 E QC-QMI  : linux_qmi_qmux_io_wake_unlock: Err in writing wakelock=qmuxd_port_wl_0, error [1:Operation not permitted]
10-30 20:01:07.343  7981  8009 W cr_ChildProcessConn: onServiceDisconnected (crash or killed by oom): pid=8416
10-30 20:01:07.352  7981  8009 I cr_BindingManager: onTrimMemory: level=20, size=0
10-30 20:01:07.520  7981  8480 W InvalidationClientCore: [TiclService] Ticl being stopped: Client: <ApplicationClientIdP: client_type=1018 client_name=b0325093-bd51-4fc2-943a-0a15ab296806>, \n\006\n\004\010\003\020\001\022\022\t\330\034\033r0gr\275\021\361J}T\325\224\035\303\032\007\010\212\"\020\003\030\001, <RunState: 2>
10-30 20:01:07.690  3726  3817 E QC-QMI  : linux_qmi_qmux_io_wake_lock: Err in writing wakelock=qmuxd_port_wl_0, error [1:Operation not permitted]
10-30 20:01:07.690  3726  3817 E QC-QMI  : linux_qmi_qmux_io_wake_unlock: Err in writing wakelock=qmuxd_port_wl_0, error [1:Operation not permitted]
10-30 20:01:07.776  7981  8009 W cr_ChildProcLH: Create a new ChildConnectionAllocator with package name = com.android.chrome, sandboxed = true
10-30 20:01:07.800  4618 10839 I ActivityManager: Start proc 8484:com.android.chrome:sandboxed_process0/u0i320 for webview_service com.android.chrome/org.chromium.content.app.SandboxedProcessService0
10-30 20:01:07.800  8484  8484 E /system/bin/webview_zygote32: Failed to make and chown /acct/uid_99320: Permission denied
10-30 20:01:07.800  8484  8484 E Zygote  : createProcessGroup(99320, 0) failed: Permission denied
10-30 20:01:07.919  8499  8499 E asset   : setgid: Operation not permitted
10-30 20:01:08.012  8484  8484 I cr_ChildProcessService: Creating new ChildProcessService pid=8484
10-30 20:01:08.045  8484  8500 E WebViewLibraryLoader: can't load with relro file; address space not reserved
10-30 20:01:08.046  8484  8500 I cr_LibraryLoader: Time to load native libraries: 8 ms (timestamps 6658-6666)
10-30 20:01:08.048  8484  8500 I cr_LibraryLoader: Using linker: org.chromium.base.library_loader.ModernLinker
10-30 20:01:08.050  8484  8500 I chromium: [INFO:library_loader_hooks.cc(136)] Chromium logging enabled: level = 0, default verbosity = 0
10-30 20:01:08.050  8484  8500 I cr_LibraryLoader: Expected native library version number "61.0.3163.98", actual native library version number "61.0.3163.98"
10-30 20:01:08.262  7981  7981 W cr_ContextualSearch: Warning, the UNDEFINED state was aborted.
10-30 20:01:08.298  4618  8365 I OpenGLRenderer: Initialized EGL, version 1.4
10-30 20:01:08.298  4618  8365 D OpenGLRenderer: Swap behavior 2
10-30 20:01:08.498  4857  4857 W AddInstrumentAdapter: Handoff activity for add instrument for transactions is disabled with a flag
10-30 20:01:08.498  4857  4857 W PaymentsAuthAdapter: Handoff activity for payments auth is disabled with a flag
10-30 20:01:08.498  4857  4857 W CvcChallengeAdapter: Handoff activity for CVC challenge for transactions is disabled with a flag
10-30 20:01:08.498  4857  4857 W FixInstrumentAdapter: Handoff activity for fix instrument for transactions is disabled with a flag
10-30 20:01:08.550  4618  4632 I WifiService: getConnectionInfo uid=10029
10-30 20:01:08.552  4618 21857 I WifiService: getWifiEnabledState uid=10029
10-30 20:01:08.552  4857  4951 W Settings: Setting mobile_data has moved from android.provider.Settings.Secure to android.provider.Settings.Global.
10-30 20:01:08.556  4857  4908 W CronetHttpEngine: Upload request without a content type.
10-30 20:01:08.558  4618 10842 I WifiService: getConnectionInfo uid=10029
10-30 20:01:08.559  4618 10844 I WifiService: getWifiEnabledState uid=10029
10-30 20:01:08.560  4857  7966 W Settings: Setting mobile_data has moved from android.provider.Settings.Secure to android.provider.Settings.Global.
10-30 20:01:08.629   437   437 D SurfaceFlinger: duplicate layer name: changing com.google.android.googlequicksearchbox/com.google.android.apps.gsa.assistant.settings.AssistantSettingsActivity to com.google.android.googlequicksearchbox/com.google.android.apps.gsa.assistant.settings.AssistantSettingsActivity#1
10-30 20:01:08.832  4618  4627 I zygote64: NativeAlloc concurrent copying GC freed 81969(4MB) AllocSpace objects, 28(1540KB) LOS objects, 42% free, 22MB/40MB, paused 789us total 152.738ms
10-30 20:01:09.411  7981  8511 W InvalidationClientCore: [TiclService] Ticl being stopped: Client: <ApplicationClientIdP: client_type=1018 client_name=b0325093-bd51-4fc2-943a-0a15ab296806>, \n\006\n\004\010\003\020\001\022\022\t\330\034\033r0gr\275\021\361J}T\325\224\035\303\032\007\010\212\"\020\003\030\001, <RunState: 2>
10-30 20:01:09.790  4857  4948 W S3NetworkUtils: [Upload] response code: 500
10-30 20:01:09.796  4857  4948 W CronetHttpEngine: Error code for a QUIC protocol, no error provided.
10-30 20:01:09.801  4857  4857 W ASPrefControlBase: #onFailure
10-30 20:01:09.801  4857  4857 W ASPrefControlBase: com.google.android.apps.gsa.shared.speech.b.o: errorCode: -83903, engine: 2
10-30 20:01:09.801  4857  4857 W ASPrefControlBase: 	at com.google.android.apps.gsa.s3.u.b(SourceFile:7)
10-30 20:01:09.801  4857  4857 W ASPrefControlBase: 	at com.google.android.apps.gsa.s3.p.a(SourceFile:9)
10-30 20:01:09.801  4857  4857 W ASPrefControlBase: 	at com.google.android.apps.gsa.s3.n.MD(SourceFile:63)
10-30 20:01:09.801  4857  4857 W ASPrefControlBase: 	at com.google.android.apps.gsa.s3.m.run(SourceFile:19)
10-30 20:01:09.801  4857  4857 W ASPrefControlBase: 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:457)
10-30 20:01:09.801  4857  4857 W ASPrefControlBase: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
10-30 20:01:09.801  4857  4857 W ASPrefControlBase: 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:457)
10-30 20:01:09.801  4857  4857 W ASPrefControlBase: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
10-30 20:01:09.801  4857  4857 W ASPrefControlBase: 	at com.google.android.apps.gsa.shared.util.concurrent.a.ag.run(Unknown Source:4)
10-30 20:01:09.801  4857  4857 W ASPrefControlBase: 	at com.google.android.apps.gsa.shared.util.concurrent.a.bo.run(SourceFile:4)
10-30 20:01:09.801  4857  4857 W ASPrefControlBase: 	at com.google.android.apps.gsa.shared.util.concurrent.a.bo.run(SourceFile:4)
10-30 20:01:09.801  4857  4857 W ASPrefControlBase: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
10-30 20:01:09.801  4857  4857 W ASPrefControlBase: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
10-30 20:01:09.801  4857  4857 W ASPrefControlBase: 	at java.lang.Thread.run(Thread.java:764)
10-30 20:01:09.801  4857  4857 W ASPrefControlBase: 	at com.google.android.apps.gsa.shared.util.concurrent.a.ak.run(SourceFile:6)
```
Here is the logcat log, sorry if any personal trash was mixed inside and great thanks for the help.

**Comment by dziedzic99** [comment]
On server side I get a POST /token request answered with code 200, and then nothing.
Can someone say whether it's Google side fault or mine?

**Comment by Fleker** [comment]
I have found where the error code -83903 is sent, although I haven't completely identified why it is being sent.

**Comment by dziedzic99** [comment]
Do you need any extra data I can provide?

**Comment by dziedzic99** [comment]
Interesting update:
I returned from holiday in Poland where my servers are located.
Everything works ok in the UK. Now I remember that it stopped working exactly when I went to Poland.
Nothing else changed beside user country.

**Comment by Fleker** [comment]
Let's use issue #58 to continue the discussion

**Comment by sunjunlishi** [comment]
recently，it is ok

**Comment by antoniosap** [comment]
no solved

**Comment by Fleker** [comment]
I think this issue is becoming too broad. The error message is generic and can be from a number of issues between the user starting the account linking to the server responding to the SYNC request. It is also difficult to give good guidance without providing the associated logs (even the logs themselves are moderately helpful).

**Comment by orcema** [comment]
i resolved my different login issues with this bug fix #108, maybe it can help someone else
